### PR TITLE
Adds (optional) support for pool param to /usage

### PR DESCRIPTION
**Pull request by dposada**
## Changes proposed in this PR

- introducing the "pool" concept described in issue #755 
- adding the pool-related schema
- adding `/pools` for fetching all pools (`GET` only)
- adding an optional `pool` query param to `/usage`
- adding support for filtering by pool in `util/get-user-running-job-ents`

## Why are we making these changes?

For full context, take a look at issue #755. We're going to be adding support for pools incrementally, without breaking backwards compatibility. This is the first increment.

### CODE REVIEW

File: integration/travis/run_integration.sh [hunk 1]
```diff
@@ -76,46 +76,53 @@ $(./minimesos info | grep MINIMESOS)
...
 # Start three cook schedulers. We want one cluster with two cooks to run MasterSlaveTest, and a second cluster to run MultiClusterTest.
 # The basic tests will run against cook-framework-1
-cd ${TRAVIS_BUILD_DIR}/scheduler
+cd ${SCHEDULER_DIR}
 ## on travis, ports on 172.17.0.1 are bindable from the host OS, and are also
 ## available for processes inside minimesos containers to connect to
 export COOK_EXECUTOR_COMMAND=${COOK_EXECUTOR_COMMAND}
 # Start one cook listening on port 12321, this will be the master of the "cook-framework-1" framework
-LIBPROCESS_IP=172.17.0.1 COOK_DATOMIC="datomic:free://localhost:4334/cook-jobs" COOK_PORT=12321 COOK_FRAMEWORK_ID=cook-framework-1 COOK_LOGFILE="log/cook-12321.log" lein run ${PROJECT_DIR}/travis/${CONFIG_FILE} &
+COOK_DATOMIC_URI_1=datomic:free://localhost:4334/cook-jobs
```
[the remark this answers is not in the document]
Comment (dposada) [reply integration/travis/run_integration.sh]: I added one. The second cluster that we start up uses its own (for now, in-memory) datomic.

File: scheduler/bin/run-docker.sh [hunk 2]
```diff
@@ -135,12 +135,11 @@ docker create \
...
-#
-#    --entrypoint=/bin/bash \
-#
-# This will override the ENTRYPOINT baked into the Dockerfile
-# and instead give you an interactive bash shell.
+docker start ${NAME}
+
+echo "Seeding test data..."
+cd ${SCHEDULER_DIR}
+lein exec -p test-resources/data/seed_pools.clj ${COOK_DATOMIC_URI}
```
[the remark this answers is not in the document]
Comment (dposada) [reply scheduler/bin/run-docker.sh]: Right now, if `COOK_DATOMIC_URI` is unset we are defaulting it to `"datomic:free://localhost:4334/cook-jobs"`, assuming that the caller is using the datomic being fired up inside the container.

```bash
if [ -z "${COOK_DATOMIC_URI}" ];
then
    COOK_DATOMIC_URI="datomic:free://localhost:4334/cook-jobs"
fi
```

File: scheduler/src/cook/mesos/api.clj [hunk 3]
```diff
@@ -2285,6 +2288,32 @@
...
+
+(defn pools-handler
+  "Handler for retrieving pools"
+  []
+  (base-cook-handler
+    {:allowed-methods [:get]
+     :handle-ok (fn [_]
+                  (let [db (d/db datomic/conn)]
+                    (->> (pool/all-pools db)
+                         (mapv (partial pool-entity->consumable-map)))))}))
```
[the remark this answers is not in the document]
Comment (dposada) [reply scheduler/src/cook/mesos/api.clj]: Done, thanks

File: scheduler/bin/run-docker.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (pschorf) [reply scheduler/bin/run-docker.sh]: Ah sorry, I didn't expand the entire file.

**Pull request merged**